### PR TITLE
Lombok support in Atlas

### PR DESCRIPTION
**Pull request by yazad3**
### Description:

Added support for [Project Lombok](https://projectlombok.org/) in Atlas. This will allow developers to avoid boilerplate code like getters/setters, constructors, builders, and delegations. This will also help improve test coverage and avoid duplication and related code smells.

### Potential Impact:

The dependency is added at [Compile Only](https://blog.gradle.org/introducing-compile-only-dependencies) so once the compiler is done processing the annotations the dependency will evaporate.

There is no change needed for using these annotations beyond the changes I have added. Your IDE should support Java Annotation Processing (both IntelliJ and Eclipse support it). You may want to add the _optional_ Project Lombok plugin ([Eclipse](https://projectlombok.org/setup/eclipse) and [IntelliJ IDEA](https://projectlombok.org/setup/intellij)) for additional refactoring.

Lombok Screenshot

### Unit Test Approach:

Added a sample sanity test.

### Test Results:

Shows use of [Getter](https://projectlombok.org/features/GetterSetter), [Builder](https://projectlombok.org/features/Builder), [Delegate](https://projectlombok.org/features/experimental/Delegate) among other use cases.

------

In doubt: [Contributing Guidelines](CONTRIBUTING.md)

### CODE REVIEW

File: src/test/java/org/openstreetmap/atlas/dependency/LombokTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * Needs annotation processing, For IntelliJ IDEA go to Preferences | Build, Execution, Deployment |
+ * Compiler | Annotation Processors and then "Enable Annotation Processing".
+ *
+ * @author Yazad Khambata
+ */
+public class LombokTest
+{
+
```
Comment (lucaspcram) [inline src/test/java/org/openstreetmap/atlas/dependency/LombokTest.java]: nit: extra line

Comment (yazad3) [reply src/test/java/org/openstreetmap/atlas/dependency/LombokTest.java]: Done thank you!

**Comment by yazad3** [comment]
Thanks @lucaspcram! Addressed the white space.

**Comment by yazad3** [comment]
@matthieun - I created this PR so its easy to see what is being done to add lombok.

In #412, there were code smells that were difficult to get rid of without the features I used from lombok.

If you are merging #412, #413 will get merged automatically.

**Pull request merged**